src/app/components/FilterPopup/FilterPopup.jsx

```diff
@@ -135,8 +136,15 @@ class FilterPopup extends React.Component {
    */
   onDateFilterChange(filterId, value) {
     const selectedFilters = this.state.selectedFilters;
+    let datePrefix = '';
 
-    selectedFilters[filterId] = value;
+    if (filterId === 'dateAfter') {
+      datePrefix = 'after ';
+    } else if (filterId === 'dateBefore') {
+      datePrefix = 'before ';
+    }
+
+    selectedFilters[filterId] = { value: `${datePrefix}${value}` };
```

Comment:
This does not make sense. You are adding an object `{ value: ... }` to `selectedFilters[filterId]` but you originally have it set up as `dateAfter: ''`. Doing this will make it `dateAfter: { value: ... }`. This should be consistent.